lib/pages/movies_page.dart

```diff
@@ -0,0 +1,162 @@
+import 'package:flutter/material.dart';
+import 'package:movie_app_jsalinas/pages/details_page.dart';
+import 'package:movie_app_jsalinas/utils/movie_repository.dart';
+import 'package:movie_app_jsalinas/widgets/movie_poster.dart';
+import 'package:movie_app_jsalinas/widgets/movie_score.dart';
+import '../../../utils/ui_constants.dart';
+import '../../../models/movie.dart';
+
+class MoviesPage extends StatelessWidget {
+  static const String routeName = 'movies/';
+  const MoviesPage({
+    super.key,
+  });
+
+  static const double borderRadius = 4;
+  static const double rowPadding = 15;
+  static const double posterHeight = 150 * 1.3;
+  static const double posterWidth = 100 * 1.3;
+  static const double imageSpacing = 20;
+  static const double textSpacing = 5;
+  static const double titleFontSize = 20;
+  static const double scoreIconScale = 0.7;
+  static const int displayedGenresAmount = 3;
+  static const double genreCardPadding = 2;
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+        future: MovieRepository().getMovies(),
+        builder: (BuildContext context, AsyncSnapshot snapshot) {
+          if (snapshot.connectionState == ConnectionState.waiting) {
+            return const CircularProgressIndicator();
+          }
+
+          if (snapshot.hasData) {
+            List<Movie> movies = snapshot.data as List<Movie>;
+            movies.add(Movie.fromStatic());
+
+            return Scaffold(
+              backgroundColor: UIConstants.detailsBackgroundColor,
+              body: SafeArea(
+                child: ListView(
+                  children: [
+                    for (Movie movie in movies)
+                      InkWell(
+                        onTap: () {
+                          Navigator.push(
+                            context,
+                            MaterialPageRoute(
+                              builder: (context) => DetailsPage(
+                                movie: movie,
+                              ),
+                            ),
+                          );
+                        },
+                        child: buildMovieCard(movie),
+                      ),
+                  ],
+                ),
+              ),
+            );
+          }
+          return const Text('no data');
+        });
+  }
+
+  Card buildMovieCard(Movie movie) {
```

Comment:
the build method, should be the last on the file, move the bellow methods above it

Reply:
solved